src/Core/Configuration.php

```diff
@@ -0,0 +1,50 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.2.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Core;
+
+/**
+ * Read-only wrapper for configuration data
+ *
+ * Intended for use with Cake\Core\Container as
+ * a typehintable way for services to have application
+ * confiugration injected as arrays cannot be typehinted.
+ */
+class Configuration
```

Comment:
This seems like a potential source of confusion for users.

Reply:
`ConfigurationContainer` maybe?

Reply:
Wouldnt it be ContainerConfiguration then maybe?

Reply:
Instead of just trying to rename Configure (which is the confusing part), we could create a `Container\ConfigureSource` that wraps Configure in in a similar way.

Reply:
I agree the name is awkward right now. I would like to keep it somewhat short though as I suspect people will be referencing this quite a bit in their service signatures.